src/server/services/procedures/hurricane-data/hurricane-data.js

```diff
@@ -8,48 +8,121 @@
 // data files are available at https://www.nhc.noaa.gov/data/ under "Best Track Data (HURDAT2)"
 // grab both the "Atlantic hurricane database" and "Northeast and North Central Pacific hurricane database" files
-const dataFiles = [
+const FALLBACK_FILES = [
     'hurdat2-1851-2020-052921.txt',
     'hurdat2-nepac-1949-2020-043021a.txt',
 ];
 
-const HurricaneData = {};
-HurricaneData._data = [];
+const DATA_ROOT = 'https://www.nhc.noaa.gov/data/hurdat';
+
+// maximum lifetime of any given downloaded resource.
+// downloads are cached for fast reuse, but will be discarded after this amount of time (milliseconds).
+const DATA_SOURCE_LIFETIME = 1 * 24 * 60 * 60 * 1000; // 1 day
 
-(function() {  // Load the hurricane data from the files
+function parseData(raw) {  // Load the hurricane data from the files
     let name = '';
+    const res = [];
     const parseLine = function (type, line) {
         if (line.startsWith('AL') || line.startsWith('EP')){
             name = line.substring(19, 28).trim();
-        } else {
-            let year = +line.substring(0, 4);
-            let month = line.substring(4,6);
-            let day = line.substring(6,8);
-            let time = line.substring(10,14);
-            let recordID = line.substring(15,17);
-            let status = line.substring(18,21).trim();
-            let latitude = line.substring(23,27);
-            let longitude = '-' + line.substring(30,35).trim();
-            let maxWind = line.substring(39,41);
-            let minPressure = line.substring(43,47).trim();
-
-            HurricaneData._data.push({name, year, month, day, time, recordID, status,
-                latitude, longitude, maxWind, minPressure});
+            return; // no actual data
         }
+        
+        res.push({ name,
+            year: +line.substring(0, 4),
+            month: line.substring(4,6),
+            day: line.substring(6,8),
+            time: line.substring(10,14),
+            recordID: line.substring(15,17),
+            status: line.substring(18,21).trim(),
+            latitude: line.substring(23,27),
+            longitude: '-' + line.substring(30,35).trim(),
+            maxWind: line.substring(39,41),
+            minPressure: line.substring(43,47).trim(),
+        });
     };
 
-    dataFiles.map(name => fs.readFileSync(path.join(__dirname, name), 'utf8'))
-        .reduce((text1, text2) => text1 + text2)
-        .split('\n')
-        .forEach(line => parseLine('AL', line));
-})();
+    raw.split('\n').forEach(line => parseLine('AL', line));
+    return res;
+}
+
+async function get(url) {
+    logger.info(`requesting ${url}`);
+    const resp = await axios({ url, method: 'GET' });
+    if (resp.status !== 200) {
+        logger.error(`download failed with status ${resp.status}`);
+        throw Error(`failed to download ${url} (status ${resp.status})`);
+    }
+    logger.info('download complete');
+    return resp.data;
+}
+function latestFile(files) {
```

Comment:
If you update your file representation, you could avoid this logic in favor of something a bit simpler. For example:
```javascript
class HurricaneDataFile {
  constructor(name) {
    this.name = name;
    this.date = new Date(year, month-1, day);  // using your regex first to get these values
  }
  async data() {
    // TODO: basically your `get` function
  }
}
// Then, later:
latestFile = files.sort((a, b) => a.date < b.date ? -1 : 1).pop();
```